source/extensions/filters/http/json_to_metadata/filter.cc

```diff
@@ -124,20 +134,47 @@ absl::flat_hash_set<std::string> FilterConfig::generateAllowContentTypes(
   return allow_content_types;
 }
 
+Regex::CompiledMatcherPtr FilterConfig::generateAllowContentTypeRegexs(
+    const envoy::type::matcher::v3::RegexMatcher& proto_allow_content_types_regex) const {
+
+  Regex::CompiledMatcherPtr allow_content_types_regex;
+  allow_content_types_regex = Regex::Utility::parseRegex(proto_allow_content_types_regex);
+
+  return allow_content_types_regex;
+}
+
 bool FilterConfig::requestContentTypeAllowed(absl::string_view content_type) const {
   if (content_type.empty()) {
     return request_allow_empty_content_type_;
   }
 
-  return request_allow_content_types_.contains(content_type);
+  if (request_allow_content_types_.contains(content_type)) {
```

Comment:
nit:
You can just return the expression:
`request_allow_content_types_.contains(content_type) || (request_allow_content_types_regex_ && request_allow_content_types_regex_->match(content_type))`

here and below vs having it eval to true to return true